test/unit/ids.js

```diff
@@ -16,26 +17,31 @@ const mockDb = (idFilterLogicFn) => {
 };
 
 exports.tearDown = callback => {
-    testUtils.restore([
-        Math.random
-    ]);
+  testUtils.restore([
+    Math.random
+  ]);
 
-    callback();
+  callback();
 };
 
-exports['generates an id of the given length'] = function(test) {
-    [1,2,3,4,5,6,7,8,9,10].forEach(function(l) {
-        test.equal(ids._generate(l).length, l);
-    });
+exports['generates an id of the given length'] = test => {
+  [1, 2, 3, 4, 5, 6, 7, 8, 9, 10].forEach(l =>
+    test.equal(ids._generate(l).length, l));
 
-    test.done();
+  test.done();
 };
 
-exports['ids can start with 0, will be correct length'] = function(test) {
-    sinon.stub(Math, 'random').returns(0.00001);
+exports['ids can start with 0, will be correct length'] = test => {
+  sinon.stub(Math, 'random').returns(0.00001);
 
-    test.equal(ids._generate(5), '00000');
-    test.done();
+  test.equal(ids._generate(5), '00000');
+  test.done();
+};
+
+exports['ids are "always" the length they should be'] = test => {
+  test.ok(
+    jsc.checkForall(jsc.integer(5, 13), i => ids._generate(i).length === i));
```

Comment:
@mandric so that is definitely *half* true!

Basically, the difference is:
 - In the normal test, I am specifically knowledgable of a potential bug in an id generation implementation where if the id starts with 0 and you make the mistake to treat the id as a number, you will loose the 0s on the front. I am then **hard-coding my testing of this known edge case**
 - In the jsverify / quickcheck / property based check / generative test (first two are libraries, second two are names for basically the same thing, for your later googling) I am **declaring a property that this algorithm has**. It is then up to the library to exercise and properly test that library. This test should catch the specific edge case I know about, but it could catch others as well.

Here is a better example: let's say I'm writing a sort algorithm. There are two obvious general properties of any sort algorithm:
 - `items.length === sorted(items).length`
 - `items[i] <= items[i++]`

There may be more, either general, or specific to your case.

You can test those a few ways:
 - Hard code some examples and presume they cover the edge cases
 - Have a test that randomly generates N collections of items (presumably also at random lengths) and then tests that each one has those properties
 - Use a QuickCheck variant to declare those properties and let the library take care of the testing

In theory each of these ideas is better and more encompassing than the previous. Good implementations of QuickCheck should fail *consistently* (as opposed to just be a flaky test), and should also attempt to narrow down the failures into the most concise (read: smallest data structure) set of inputs that cause the failure.

Again, I'm not 100% sold but I'm interested in giving it a try, and again it's probably not something you can apply everywhere, and more something that is a specific tool for a specific purpose.